resqpy/grid/write_nexus_corp.py

```diff
@@ -43,30 +44,29 @@ def write_nexus_corp(grid,
         log.warning('ijk handedness not known')
     elif not ijk_right_handed:
         log.warning('ijk axes are left handed; inverted (fake) xyz handedness required')
-    crs_root = grid.extract_crs_root()
+    crs_uuid = grid.crs_uuid
     if not local_coords:
         if not global_z_increasing_downward:
             log.warning('global z is not increasing with depth as expected by Nexus')
             tm.log_nexus_tm('warning')
-        if crs_root is not None:  # todo: otherwise raise exception?
+        if crs_uuid is not None:  # todo: otherwise raise exception?
             log.info('converting corner points from local to global reference system')
-            grid.local_to_global_crs(cp,
-                                     crs_root,
-                                     global_xy_units = global_xy_units,
-                                     global_z_units = global_z_units,
-                                     global_z_increasing_downward = global_z_increasing_downward)
+            cp = grid.local_to_global_crs(cp,
+                                          crs_uuid = crs_uuid,
+                                          global_xy_units = global_xy_units,
+                                          global_z_units = global_z_units,
+                                          global_z_increasing_downward = global_z_increasing_downward)
     log.info('writing simulator corner point file ' + file_name)
     with open(file_name, 'w') as header:
         header.write('! Nexus corner point data written by resqml_grid module\n')
         header.write('! Nexus is a registered trademark of the Halliburton Company\n\n')
+        crs = rqc.Crs(grid.model, uuid = crs_uuid)
         if write_units_keyword:
             if local_coords:
-                if crs_root is not None:
-                    crs_xy_units_text = rqet.find_tag(crs_root, 'ProjectedUom').text
-                    crs_z_units_text = rqet.find_tag(crs_root, 'VerticalUom').text
-                    if crs_xy_units_text == 'm' and crs_z_units_text == 'm':
+                if crs_uuid is not None:
+                    if crs.xy_units == 'm' and crs.z_units == 'm':
                         header.write('METRIC\n\n')
-                    elif crs_xy_units_text == 'ft' and crs_z_units_text == 'ft':
+                    elif crs.xy_units == 'ft' and crs.z_units == 'ft':
```

Comment:
please change this to ...units.startswith('ft') in both instances – this will allow acceptance of 'ft[US]' etc.